Distributed cargo spawner

CrystalineSpider opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
Resolves #356

No mapping changes yet, just the spawner and a selection of crates for it.

Later we can make it weighted, but I wanted to at least get the core pushed.
## Media
<img width="283" height="634" alt="Screenshot 2025-12-22 174019" src="https://github.com/user-attachments/assets/126e00d9-16f2-44ee-8e4c-55dd530c8548" />
<img width="240" height="595" alt="Screenshot 2025-12-22 174036" src="https://github.com/user-attachments/assets/8cb2d1e1-ba0f-412a-ab60-1ea6876c99ac" />

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an in-game showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

CrystalineSpider committed:
Adds cargo spawns and crates for it

CrystalineSpider committed:
fixing typo

CrystalineSpider committed:
more typos

EmoGarbage404 commented on Resources/Prototypes/_ES/Catalog/Crates/crates.yml:
```diff
@@ -0,0 +1,279 @@
+  - type: EntityTableContainerFill
+    containers:
+      entity_storage: !type:AllSelector
+        children:
```
```suggestion
      entity_storage:
        all:
```
You can do this to shorten it a bit and make it easier to read (this can be done across most of this file)

EmoGarbage404 commented on Resources/Prototypes/_ES/Entities/Markers/Spawners/Distributed/cargo.yml:
```diff
@@ -0,0 +1,50 @@
+  - type: Sprite
+    layers:
+    - state: crate
+    - state: random
```
```suggestion
    - state: rand-dist
```

as a distributed spawner, should use the `rand-dist` state instead of `random`

EmoGarbage404 commented on Resources/Prototypes/_ES/Entities/Markers/Spawners/Distributed/cargo.yml:
```diff
@@ -0,0 +1,50 @@
+  components:
+  - type: Sprite
+    layers:
+    - state: crate
```
the crate icon is fairly generic so you should add a new sprite to the spawner rsi that says "CARGO" or something so that they're easier to distinguish. There's a ton of crate spawners and having it denoted as specifically for cargo would improve readability when mapping.

CrystalineSpider replied:
I wasn't sure how the sprites worked so left the crate as mostly a default, I'll work on a proper label though since I do agree

EmoGarbage404 reviewed (changes requested):
didnt really look at the crate fills beyond a glance. we could use more coverage but that's easy to add onto once we have this part generally ironed out.

CrystalineSpider commented:
> didnt really look at the crate fills beyond a glance. we could use more coverage but that's easy to add onto once we have this part generally ironed out.

What do you mean?

EmoGarbage404 commented:
> What do you mean?

there's a lot more potential crates we could add, but I think this is probably fine for a first selection since we're really just hashing out the basics for now

CrystalineSpider commented:
> there's a lot more potential crates we could add, but I think this is probably fine for a first selection since we're really just hashing out the basics for now

That was mostly my thought. I haven't played much cargo and don't really know what people need/want but I figured we could always do a second pass once the framework is there.

CrystalineSpider committed:
polish, weights, and resolving issues.

CrystalineSpider commented:
added weights to the items so the new sec crates wouldn't make up the entire cargo bay and resolved your requested changes. Also added a few crates, mostly various security crates and a few other lower weight ones.

EmoGarbage404 committed:
Nothing change

EmoGarbage404 commented on Resources/Prototypes/_ES/Entities/Markers/Spawners/Distributed/cargo.yml:
```diff
@@ -0,0 +1,74 @@
+    - state: c100
+  - type: EntityTableSpawner
+    table: !type:NestedSelector
+      tableId: ESSpawnerCargo
```
this isn't using a distributed spawner anymore, which needs to be fixed. it should look something like

```yml
  - type: ESDistributedSpawner
    id: ESSpawnerCargo
    tables:
    - tableId: ESSpawnerCargo
    - tableId: ESSpawnerCargo
    - tableId: ESSpawnerCargo
    - tableId: ESSpawnerCargo
    - tableId: ESSpawnerCargo
 ```

realistically in the future we should probably weight this so it's less "20 fully random crate spawners" and more like "2 of this group, 5 of this group, 10 of this group, etc."

CrystalineSpider replied:
If I may ask, whats the purpose of having the tableId replicated?

Also I agree doing weights with certain amounts of various crate sets would be ideal, and while I might look into that later it's functional enough in my opinion as is. (Most of this was based off of me seeing examples of functions in other yaml files and I just didn't find a good example of that kind of randomization, if I find one though I will likely return to revise this again)

CrystalineSpider replied:
Actually I think I figured out the way to weigh this to have less random options so I'll get back to this.

CrystalineSpider committed:
Proper randomization, final recode I swear

CrystalineSpider committed:
minor changes

CrystalineSpider committed:
Merge branch 'dist-cargo-spawners' of https://github.com/CrystalineSpider/ephemeral-space into dist-cargo-spawners

CrystalineSpider commented:
I hope this works, changes TLDR because I feel formal, but spawner now spawns a set amount of each "department" of crate, such as four material-related crates or two medical-related crates.

CrystalineSpider committed:
typo fix (also minor last minute weight change because I'm not feeling it anymore)

EmoGarbage404 committed:
Mild fixes
